src/adagrams.js

```diff
@@ -1,15 +1,157 @@
+const letterPool = {
+    'A': 9,
+    'B': 2,
+    'C': 2,
+    'D': 4,
+    'E': 12,
+    'F': 2,
+    'G': 3,
+    'H': 2,
+    'I': 9,
+    'J': 1,
+    'K': 1,
+    'L': 4,
+    'M': 2,
+    'N': 6,
+    'O': 8,
+    'P': 2,
+    'Q': 1,
+    'R': 6,
+    'S': 4,
+    'T': 6,
+    'U': 4,
+    'V': 2,
+    'W': 2,
+    'X': 1,
+    'Y': 2,
+    'Z': 1
+}
+
+// need to convert #'s back to int
+
+const scoreDict = {
+    A: 1,
+    B: 3,
+    C: 3,
+    D: 2,
+    E: 1,
+    F: 4,
+    G: 2,
+    H: 4,
+    I: 1,
+    J: 8,
+    K: 5,
+    L: 1,
+    M: 3,
+    N: 1,
+    O: 1,
+    P: 3,
+    Q: 10,
+    R: 1,
+    S: 1,
+    T: 1,
+    U: 1,
+    V: 4,
+    W: 4,
+    X: 8,
+    Y: 4,
+    Z: 10,
+}
+const alphabet = "abcdefghijklmnopqrstuvwxyz".toUpperCase().split("");
+
+
+
+
+
+// Implement this method for wave 1
 export const drawLetters = () => {
-  // Implement this method for wave 1
+    const allLetters = []
+    for (const [letter, frequency] of Object.entries(letterPool)) {
+        for (let i = 0; i < frequency; i++) {
+            allLetters.push(letter)
+        }
+    }
+
+    const hand = []
+
+    for (let i = 0; i < 10; i++) {
+        const thisLetter = allLetters[Math.floor(Math.random() * allLetters.length)]
+        hand.push(thisLetter)
+        let letterIndex = allLetters.indexOf(thisLetter)
+        allLetters.splice(letterIndex, 1)
```

Comment:
Remember that removing items from an array in O(n). For an arbitrary number of tiles, this would be a O(n^2) operation (but since we're artificially restricting to 10 tiles, we can still say this is O(10n) → O(n)). Consider shuffling the array and then picking any contiguous block of 10 tiles as the hand. This additive approach would remain O(n) regardless of the number of tiles we needed to pick. In general, try to avoid subtractive approaches (at least on arrays) since that will trend towards O(n^2) overall approaches.